src/System.Private.ServiceModel/tools/test/SelfHostWcfService/TestResources/WebSocketResources.cs

```diff
@@ -118,4 +119,88 @@ protected override Binding GetBinding()
             return binding;
         }
     }
+
+    internal class WebSocketHttpsDuplexTextStreamedResource : EndpointResource<WSDuplexService, IWSDuplexService>
+    {
+        protected override string Protocol { get { return BaseAddressResource.Https; } }
+
+        protected override int GetPort(ResourceRequestContext context)
+        {
+            return context.BridgeConfiguration.BridgeHttpsPort;
+        }
+
+        protected override string Address { get { return "WebSocketHttpsDuplexTextStreamedResource"; } }
+
+        protected override Binding GetBinding()
+        {
+            int DefaultMaxReceivedMessageSize = 64 * 1024 * 1024;
+
+            TextMessageEncodingBindingElement textMessageEncodingBindingElement = new TextMessageEncodingBindingElement();
+            HttpsTransportBindingElement httpsTransportBindingElement = new HttpsTransportBindingElement();
+            httpsTransportBindingElement.WebSocketSettings.TransportUsage = WebSocketTransportUsage.Always;
+            httpsTransportBindingElement.MaxReceivedMessageSize = DefaultMaxReceivedMessageSize;
+            httpsTransportBindingElement.MaxBufferSize = DefaultMaxReceivedMessageSize;
+            httpsTransportBindingElement.TransferMode = TransferMode.Streamed;
```

Comment:
NIT: not required - consider using 

```csharp
HttpsTransportBindingElement httpsTransportBindingElement = new HttpsTransportBindingElement() 
{
    MaxReceivedMessageSize = DefaultMaxReceivedMessageSize, 
    TransferMode = TransferMode.Streamed
    ... 
}
```

style constructor to make this look a little nicer.